nx/lib/nx/lin_alg.ex

```diff
@@ -1422,7 +1421,23 @@ defmodule Nx.LinAlg do
        %{tensor | names: List.duplicate(nil, rank - 1), type: output_type, shape: s_shape},
        %{tensor | names: List.duplicate(nil, rank), type: output_type, shape: v_shape}}
 
-    Nx.Shared.optional(:svd, [tensor, opts], output, &Nx.LinAlg.SVD.svd/2)
+    {u, s, vt} =
+      Nx.Shared.optional(:svd, [tensor, opts], output, fn tensor, opts ->
+        {u, s, vt} = Nx.LinAlg.SVD.svd(tensor, opts)
+
+        # THIS IS A BUG, DO NOT MERGE AS IS
+        {
+          Nx.devectorize(u, keep_names: false),
+          Nx.devectorize(s, keep_names: false),
+          Nx.devectorize(vt, keep_names: false)
+        }
```

Comment:
Somehow the tuple returned from the SVD defn call is not devectorized, so we have at least this bug and the one I'll point out below to prune before merging